src/utils.jl

```diff
@@ -54,3 +54,53 @@ function checkbalanced(x::AbstractVecOrMat, y::AbstractVecOrMat)
         throw(ArgumentError("source and target marginals are not balanced"))
     return nothing
 end
+
+struct FiniteDiscreteMeasure{X<:AbstractArray,P<:AbstractVector}
+    support::X
+    p::P
+
+    function FiniteDiscreteMeasure{X,P}(support::X, p::P) where {X,P}
+        size(support, 1) == length(p) ||
+            error("number of rows of `support` and `p` must be equal")
+        sum(p) ≈ 1 || error("`p` must sum to 1")
+        all(p .>= 0) || error("values of `p` must be greater of equal than 0")
+        return new{X,P}(support, p)
+    end
+end
+
+"""
+    FiniteDiscreteMeasure(support::AbstractArray, p::AbstractVector)
+Construct a finite discrete probability measure with support `support` and corresponding weights `p`.
+"""
+function FiniteDiscreteMeasure(support::AbstractArray, p::AbstractVector)
+    P = sum(p)
+    if size(support, 2) == 1
+        return if P ≈ 1
+            DiscreteNonParametric(vec(support), p)
+        else
+            DiscreteNonParametric(vec(support), p ./ P)
+        end
+    else
+        return if P ≈ 1
+            FiniteDiscreteMeasure{typeof(support),typeof(p)}(support, p)
+        else
+            FiniteDiscreteMeasure{typeof(support),typeof(p)}(support, p ./ P)
+        end
+    end
+end
+
+"""
+    FiniteDiscreteMeasure(support::AbstractArray)
+Construct a finite discrete probability measure with support `support` and equal probability for each point.
+"""
+function FiniteDiscreteMeasure(support::AbstractArray)
+    p = ones(size(support)[1]) ./ size(support)[1]
+    if size(support, 2) == 1
+        return DiscreteNonParametric(vec(support), p)
+    else
+        return FiniteDiscreteMeasure{typeof(support),typeof(p)}(support, p)
+    end
+end
+
```

Comment:
Again, this is not a necessary "feature", but I'd argue it's very convenient for users. I have been doing a course on OT, so pretty much every week I have created a new notebook to showcase some OT. And as a user, I really appreciate this kind of functionality in which the functions already does most of the work for me. I guess since I come from Python, I'm mostly reckless in terms of efficiency, and I'm much more user-centric.
So, if this type of stuff creates issues, I think we should drop it. Otherwise, I'd love to have it :)

Reply:
Where is this function `Fill` from ? I just know `fill`.